webapp/channels/src/components/preparing_workspace/preparing_workspace.tsx

```diff
@@ -98,28 +107,35 @@ const PreparingWorkspace = (props: Props) => {
         defaultMessage: 'Something went wrong. Please try again.',
     });
     const isUserFirstAdmin = useSelector(isFirstAdmin);
-    const useCaseOnboarding = useSelector(getUseCaseOnboarding);
 
     const currentTeam = useSelector(getCurrentTeam);
     const myTeams = useSelector(getMyTeams);
 
     // In cloud instances created from portal,
     // new admin user has a team in myTeams but not in currentTeam.
-    const team = currentTeam || myTeams?.[0];
+    let team = currentTeam || myTeams?.[0];
 
     const config = useSelector(getConfig);
     const pluginsEnabled = config.PluginsEnabled === 'true';
     const showOnMountTimeout = useRef<NodeJS.Timeout>();
+    const configSiteUrl = config.SiteURL;
+    const isSelfHosted = useSelector(getLicense).Cloud !== 'true';
 
     const stepOrder = [
+        isSelfHosted && WizardSteps.Organization,
         pluginsEnabled && WizardSteps.Plugins,
+        isSelfHosted && WizardSteps.InviteMembers,
         WizardSteps.LaunchingWorkspace,
     ].filter((x) => Boolean(x)) as WizardStep[];
 
+    // first steporder that is not false
+    const firstShowablePage = stepOrder.filter((x) => Boolean(x))[0];
```

Comment:
Forgot to mention; this can just be `const firstShowablePage = stepOrder[0]` since the definition above already filters out all the false ones.